### PR TITLE
adds an Enum for the table header; tries to improve logic encapsulation

### PR DESCRIPTION
**Pull request by aocneanu**

### CODE REVIEW

File: src/Services/Finder.php [hunk 1]
```diff
@@ -3,50 +3,39 @@
...
 class Finder
 {
-    public static array $folders = ['.'];
-    public static array $vendors = ['laravel-enso'];
-
     public function upgrades(): Collection
     {
-        return $this->initUpgrades()
-            ->filter->hasUpgrade()
+        \Log::info($this->upgradePackages());
```
Comment (gandesc) [inline src/Services/Finder.php]: log leftover

File: src/Services/Package.php [hunk 2]
```diff
@@ -4,25 +4,30 @@
...
         $this->folder = $folder;
     }
 
-    public function isPackage(): bool
+    public function qualifies(): bool
     {
-        return File::exists($this->folder.DIRECTORY_SEPARATOR.'composer.json');
-    }
+        if ($this->hasUpgrades()) {
+            \Log::info($this->folder);
```
Comment (gandesc) [inline src/Services/Package.php]: log

File: src/Services/Package.php [hunk 3]
```diff
@@ -32,26 +37,32 @@ public function upgradeClasses()
...
+            ))->filter(fn ($class) => $this->isUpgrade($class));
     }
 
-    private function appFolder(...$parts): string
+    private function hasUpgrades(): bool
     {
-        return (new Collection([
-            $this->folder, $this->psr4Folder(), ...$parts,
-        ]))->implode(DIRECTORY_SEPARATOR);
+        \Log::info($this->upgradeFolder());
```
Comment (gandesc) [inline src/Services/Package.php]: log

**Review by raftx24 — APPROVED** [review]
Great

**Pull request merged**